src/mpi/init/init_async.c

```diff
@@ -31,12 +31,31 @@
         highly system dependent but may be substantial in some cases,
         hence this recommendation.
 
+    - name        : MPIR_CVAR_CH4_PROGRESS_THREAD_AFFINITY
+      category    : CH4
+      type        : string
+      default     : ""
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        Specifies affinity for all progress threads of local processes. Format - comma-separated list of logical processors.
+        In case of N progress threads per process the first N logical processors from list will be assigned
+        to threads of first local process, the next N logical processors from list - to second local process and so on.
+        For example, thread affinity is "0,1,2,3", 2 progress threads per process and 2 processes per node.
+        Progress threads of first local process will be pinned on logical processors "0,1",
+        progress threads of second local process - on "2,3".
+        Cannot work together with MPIR_CVAR_NUM_CLIQUES or MPIR_CVAR_ODD_EVEN_CLIQUES.
```

Comment:
The "auto" option is not in the description

Reply:
Will add it.

Reply:
I'm thinking about squashing the two commits: `ch4: support auto affinity of async progress threads` and `ch4: Set progress thread affinity`. That will reduce conflicts from the code changes at review.

Reply:
Done adding description for "auto"